python/lsst/cell_coadds/_fits.py

```diff
@@ -75,11 +110,61 @@ def __init__(self, filename: str) -> None:
 
         self.filename = filename
 
+    @classmethod
+    def isCompatibleWith(cls, written_version: str, /) -> bool:
+        """Check if the serialization version is compatible with the reader.
+
+        Notes
+        -----
+        This accepts the other version as a positional argument only.
+
+        Parameters
+        ----------
+        written_version: `str`
+            The version to compare against.
+
+        Returns
+        -------
+        compatible : `bool`
+            Whether the reader is compatible with ``written_version``.
+        """
+        this_version = version.parse(FILE_FORMAT_VERSION)
+        other_version = version.parse(written_version)
+
+        if this_version.major != other_version.major:
+            return False
+
+        if this_version.minor < other_version.minor:
+            return False
+
+        if this_version.major == 0 and this_version.minor != other_version.minor:
+            logger.warn(
+                "Minor version of the %s does not match the version on the file, for major version 0."
+                "The read is not guaranteed to be successful",
+                cls.__name__,
+            )
```

Comment:
I don't think we want the warning here; we can and should warn about specific things we're dropping on the floor, but warning when there's a good chance everything is fine is a bit too alarmist.